vizro-core/src/vizro/actions/_callback_mapping/_callback_mapping_utils.py

```diff
@@ -32,21 +32,27 @@ def _get_inputs_of_controls(page: Page, control_type: ControlType) -> list[State
     """Gets list of `States` for selected `control_type` of triggered `Page`."""
     return [
         State(component_id=control.selector.id, component_property=control.selector._input_property)
-        for control in page.controls
-        if isinstance(control, control_type)
+        for control in cast(Iterable[ControlType], model_manager._get_models(control_type, page))
```

Comment:
Ahh, that's so nice, that we can get all models of a specific type from a selected page as well! 🚀

Reply:
Yes, this has been a TODO for a long time for the `model_manager`. It's still pretty hacky tbh but will definitely improve in future.

Reply:
Does this essentially equate controls and components? Like I could put Graph in controls, and controls into components?

Reply:
I think yes, you could do that. So long as you have the right sort of model it shouldn't matter whether it's in `controls` or `components` or `navigation` or wherever.

Reply:
Coming back to this, I think it's amazing. We essentially have the control and component section as fillable canvases, and it is up to the user how to fill it. A little less design guidance, but quite a bit more flexibility ❤️

Reply:
I am not sure how much I like it tbh because then the controls vs. components becomes more about page layout rather than functionality. Like almost we are referring to these bits as `vm.Page(sidebar=[vm.Filter(...)], main=[vm.Graph(...)])` etc.? That's [quite similar to what streamlit does](https://docs.streamlit.io/develop/api-reference/layout/st.sidebar) and not necessarily a bad idea but quite different from how we currently do things. I can also see where the extra flexibility is useful here though.

Reply:
It could still be a case of guiding through argument name, but allowing more than that! Ie you _should_ put your controls in the control argument, and we won't let you even do it elsewise without a little extra work (ie `.add_component`) BUT ....